rectools/models/sasrec.py

```diff
@@ -922,18 +928,54 @@ def on_train_start(self) -> None:
 
     def training_step(self, batch: Dict[str, torch.Tensor]) -> torch.Tensor:
         """TODO"""
+        x, y, w = batch["x"], batch["y"], batch["yw"]
         if self.loss == "softmax":
-            return self._calc_softmax_loss(batch)
+            logits = self._get_logits(x)
+            return self._calc_softmax_loss(logits, y, w)
         if self.loss == "BCE":
-            return self._calc_bce_loss(batch)
+            negatives = batch["negatives"]
+            logits = self._get_pos_neg_logits(x, y, negatives)
+            return self._calc_bce_loss(logits, y, w)
         if self.loss == "gBCE":
-            return self._calc_gbce_loss(batch)
+            negatives = batch["negatives"]
+            logits = self._get_pos_neg_logits(x, y, negatives)
+            return self._calc_gbce_loss(logits, y, w, negatives)
 
         raise ValueError(f"loss {self.loss} is not supported")
 
-    def _calc_softmax_loss(self, batch: Dict[str, torch.Tensor]) -> torch.Tensor:
-        x, y, w = batch["x"], batch["y"], batch["yw"]
-        item_embs, session_embs = self.forward(x)
+    def _get_logits(self, x: torch.Tensor) -> torch.Tensor:
```

Comment:
```suggestion
    def _get_full_catalogue_logits(self, x: torch.Tensor) -> torch.Tensor:
```

Reply:
Sorry, I actually realised that we have "catalog" everywhere in framework, not "catalogue". Let's stick to "catalog"

Reply:
fixed everywhere